src/tensor.jl

```diff
@@ -0,0 +1,68 @@
+using PyCall
+using Statistics
+using TensorDecompositions
+
+"Run Parafac Factorization with Mask"
+function CP_decomposition(rank::Int64=5)
+    # Init
+    decomps = pyimport("tensorly.decomposition")
+    cube = createCube()
+    
+    # Create Mask/Zero Out Data
+    mask = .!(cube .=== nothing)
+    cube[cube .=== nothing] .= 0
+    
+    # Convert Data Types
+    cube = convert(Array{Float64,3}, cube)
+    mask = convert(Array{Bool,3}, mask)
+    
+    # Run Factorizaton
+    weights, factors = decomps.parafac(cube, rank, mask=mask)
+    return factors
+end
+
+"Run Tucker Factorization with Mask"
+function tucker_decomposition(rank::Tuple=(2,2,2))
+    # Init
+    decomps = pyimport("tensorly.decomposition")
+    cube = createCube()
+    
+    # Create Mask/Zero Out Data
+    mask = .!(cube .=== nothing)
+    cube[cube .=== nothing] .= 0
+    
+    # Convert Data Types
+    cube = convert(Array{Float64,3}, cube)
+    mask = convert(Array{Bool,3}, mask)
+    
+    # Run Factorizaton
+    core, factors = decomps.tucker(cube, rank, mask=mask)
+    return (core, factors)
+end
+
+"Re-compose tensor from CP decomposition"
+function cp_reconstruct(factors::Array)
+    lambdas = ones(1, size(factors[1], 2))
+    lambdas = vec(lambdas)
+    tup = (factors[1], factors[2], factors[3])
+    dest = ones(181, 22, 41)
+    compose!(dest, tup, lambdas)
+end
+
+"Re-compose tensor from Tucker decomposition"
+function tucker_reconstruct(output::Any)
```

Comment:
You don't need to add a type for every argument—in fact, unless you want to limit the type, it's better to leave it off. You can do this by just removing `::Any`.